Build breaks with AGP 3.6-beta01

### Expected behavior
Build finishes

### Observed behavior
Build halts with error

### Steps to reproduce
Update Android Gradle Plugin to latest 3.6 beta

### Version
4.7.0

### Additional information
output:

```
./gradlew :sonya:processBugsnagDev-debugManifest                                                                                                                                                                                                                     /c/Projects/sonya-android
> Task :sonya:processBugsnagDev-debugManifest FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':sonya:processBugsnagDev-debugManifest'.
> Cannot cast object 'property(interface org.gradle.api.file.Directory, fixed(class org.gradle.api.internal.file.DefaultFilePropertyFactory$FixedDirectory, C:\Projects\sonya-android\sonya\build\intermediates\bundle_manifest\devDebug\bundle-manifest))' with class 'org.gradle.api.internal.file.DefaultFilePropertyFactory$DefaultDirectoryVar' to class 'java.io.File'

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,8 @@
+## TBD
+
+Fix AGP 3.6.0 breaking project builds due to changed return type of `getBundleManifestOutputDirectory()`
+[#179](https://github.com/bugsnag/bugsnag-android-gradle-plugin/pull/179)
+
 ## 4.7.0 (2019-10-14)
 
 Add ability to override reported value for versionCode

--- a/build.gradle
+++ b/build.gradle
@@ -152,7 +152,7 @@ task wrapper(type: Wrapper) {
 codenarc {
     // ignore baseline violations for now
     maxPriority2Violations 7
-    maxPriority3Violations 11
+    maxPriority3Violations 10
 }
 
 // license checking

--- a/build.gradle
+++ b/build.gradle
@@ -152,7 +152,7 @@ task wrapper(type: Wrapper) {
 codenarc {
     // ignore baseline violations for now
     maxPriority2Violations 7
-    maxPriority3Violations 9
+    maxPriority3Violations 11
 }
 
 // license checking

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy
@@ -293,7 +293,7 @@ class BugsnagPlugin implements Plugin<Project> {
         if (processManifest.hasProperty("bundleManifestOutputDirectory")) {
 
             // For AGP versions >= 3.3.0 the bundle manifest is output to its own directory
-            def directory = processManifest.getBundleManifestOutputDirectory()
+            def directory = processManifest.bundleManifestOutputDirectory
 
             if (directory instanceof File) { // 3.3.X - 3.5.X returns a File
                 return directory

--- a/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy
+++ b/src/main/groovy/com/bugsnag/android/gradle/BugsnagPlugin.groovy
@@ -291,8 +291,16 @@ class BugsnagPlugin implements Plugin<Project> {
 
     static File resolveBundleManifestOutputDirectory(ManifestProcessorTask processManifest) {
         if (processManifest.hasProperty("bundleManifestOutputDirectory")) {
+
             // For AGP versions >= 3.3.0 the bundle manifest is output to its own directory
-            return processManifest.bundleManifestOutputDirectory
+            def directory = processManifest.getBundleManifestOutputDirectory()
+
+            if (directory instanceof File) { // 3.3.X - 3.5.X returns a File
+                return directory
+            } else { // 3.6.+ returns a DirectoryProperty
+                return directory.asFile.get()
+            }
+
         } else {
             // For AGP versions < 3.3.0 the bundle manifest is the merged manifest
             return processManifest.manifestOutputDirectory

